beetsplug/lyrics.py

```diff
@@ -291,14 +291,25 @@ def fetch(self, artist, title):
             self._log.warning(u'we are blocked at MusixMatch: url %s failed'
                               % url)
             return
-        html_part = html.split('<p class="mxm-lyrics__content')[-1]
-        lyrics = extract_text_between(html_part, '>', '</p>')
+        html_parts = html.split('<p class="mxm-lyrics__content')
+        # Sometimes lyrics come in 2 parts
+        if len(html_parts) > 1:
+            html_part1 = html_parts[-2]
+            lyrics_part1 = extract_text_between(html_part1, '>', '</p>')
+        else:
+            lyrics_part1 = ''
+        html_part2 = html_parts[-1]
+        lyrics_part2 = extract_text_between(html_part2, '>', '</p>')
+        lyrics = lyrics_part1 + '\n' + lyrics_part2
```

Comment:
Would it be simpler to do this with a loop over all the parts? Something roughly like this:

```py
for html_part in html_parts:
    lyrics_parts.append(extract_text_between(html_part1, '>', '</p>'))
lyrics = '\n'.join(lyrics_parts)
```

Reply:
Thanks for the correction.
Indeed, this approach is more elegant. I've updated my branch to use the loop, as well as better non-existing lyrics detection.
I've tested it on couple songs I have, and it seems to work.